### ISSUE TITLE
Custom metric for (I) skipping NA in y_true and (ii) multiple outputs

### ISSUE THREAD

**Issue by daniel-althoff** [rstudio/keras#1211]
Hi,

I am trying to build a custom LSTM to predict multiple outputs.
First, how do I set different metrics for each output? 

I have seen an example for Python:
``x1 = layers.Conv2D(3, 3)(image_input)
x1 = layers.GlobalMaxPooling2D()(x1)

x2 = layers.Conv1D(3, 3)(timeseries_input)
x2 = layers.GlobalMaxPooling1D()(x2)

x = layers.concatenate([x1, x2])

score_output = layers.Dense(1, name="score_output")(x)
class_output = layers.Dense(5, name="class_output")(x)

model = keras.Model(
    inputs=[image_input, timeseries_input], outputs=[score_output, class_output]
)

    optimizer=keras.optimizers.RMSprop(1e-3),
    loss={
        "score_output": keras.losses.MeanSquaredError(),
        "class_output": keras.losses.CategoricalCrossentropy(),
    },
    metrics={
        "score_output": [
            keras.metrics.MeanAbsolutePercentageError(),
            keras.metrics.MeanAbsoluteError(),
        ],
        "class_output": [keras.metrics.CategoricalAccuracy()],
    },
)``

Can this be done in R? Do I have to pass out two outputs?

Additionally, sometimes one of the multiple outputs may be missing, can I create a custom metric that skips evaluating the prediction when y_true is NA?

for example, cast them equal 0 and instead of calculating the mean, divide them by "N", which is the number of observations (counted) for complete observations? IDK. Any help would be great.
